### PR TITLE
Update CODEOWNERS for PyTorch/XLA team review access

### PR DESCRIPTION
**Pull request by ManfeiBai**
# Description
 add permission for PyTorch/XLA team review

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by RissyRan** [comment]
Hi Manfei, this is code owner for infra changes. Could you add your name under `dags/pytorch_xla` and/or `dags/mantaray` if applies?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # Default owners for everything in the repo, unless a later match takes precedence.
-* @mbzomowski @RissyRan @allenwang28
+* @mbzomowski @RissyRan @allenwang28 @ManfeiBai
```
Comment (ManfeiBai) [inline .github/CODEOWNERS:3]: hi, @RissyRan , do we want to remove this change?

Comment (RissyRan) [reply .github/CODEOWNERS:3]: You may want to discuss with Yixin and team to see if you are the owner of the XL ML infra. I think @richardsliu currently is the new owner.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
 dags/solutions_team/configs/tensorflow @chandrasekhard2 @ZhaoyueCheng @richardsliu
 dags/solutions_team/solutionsteam_tf* @chandrasekhard2 @ZhaoyueCheng @richardsliu
 
 dags/pytorch_xla @JackCaoG @vanbasten23 @zpcore @ManfeiBai
 dags/legacy_test/tests/pytorch @JackCaoG @vanbasten23 @zpcore @ManfeiBai
 
+dags/mantaray @ManfeiBai
```
Comment (RissyRan) [inline .github/CODEOWNERS:10]: nit: add a space after the line?

Also, you may want to add an extra folk to unblock your PR review for this folder.

File: .github/CODEOWNERS (hunk 1 above)
Comment (ManfeiBai) [reply .github/CODEOWNERS:3]: thanks, be the owner for `dags/mantaray` should cover our request for now, let's get back to this discussion later if needed

File: .github/CODEOWNERS (hunk 2 above)
Comment (ManfeiBai) [reply .github/CODEOWNERS]: Thanks, updated and added @tengyifei and @zpcore as the extra folk too

**Review by RissyRan — APPROVED** [review]

**Pull request merged**